packages/backend/src/routes/UsernameRouter.ts

```diff
@@ -69,6 +69,25 @@ async function setUsername(req, res) {
         return
     }
 
+    // check if user has already set username in current epoch before
+    const epochKey = reputationProof.epochKey.toString()
+    const records = (
+        await req.db.findMany('Record', {
+            where: {
+                from: epochKey,
+                to: epochKey,
+            },
+        })
+    ).filter(
+        (r) => r.action === ActionType.SetUsername && r.epoch === currentEpoch
+    )
```

Comment:
you can use the db query here
```ts
const records = await req.db.findMany('Record', {
    where: {
        from: epochKey,
        to: epochKey,
        action: ActionType.SetUsername,
        epoch: currentEpoch
    }
})
```

Reply:
or use `count` instead of `findMany` to get the length of matched records
```ts
const records = await req.db.count('Record', {
    from: epochKey,
    to: epochKey,
    action: ActionType.SetUsername,
    epoch: currentEpoch
})
```

and check `if (records > 0)` then return error

Reply:
> you can use the db query here
> 
> ```ts
> const records = await req.db.findMany('Record', {
>     where: {
>         from: epochKey,
>         to: epochKey,
>         action: ActionType.SetUsername,
>         epoch: currentEpoch
>     }
> })
> ```

haha you're right, i just copy from anywhere, will fix it